src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BitbucketRevisionAction.java

```diff
@@ -0,0 +1,42 @@
+package com.atlassian.bitbucket.jenkins.internal.status;
+
+import hudson.model.Action;
+
+import javax.annotation.CheckForNull;
+
+public class BitbucketRevisionAction implements Action {
+
+    private final String revisionSha1;
+    private final String serverId;
+
+    public BitbucketRevisionAction(String revisionSha1, String serverId) {
```

Comment:
This seems like a really good solution to me- it means we can remove `getLatestRevision` from the SCM entirely.